app/src/basewidget.h

```diff
@@ -0,0 +1,32 @@
+/*
+
+Pencil2D - Traditional Animation Software
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+
+#ifndef BASEWIDGET_H
+#define BASEWIDGET_H
+
+#include <QWidget>
+
+class BaseWidget : public QWidget
+{
+public:
+    BaseWidget(QWidget* parent = nullptr);
+
+    virtual void initUI();
+    virtual void updateUI();
```

Comment:
Looks ike`BaseWidget` is designed to be an interface:
I suggest to make `initUI()` and `updateUI` pure virtual functions so it will force the derived class to implement them.
```
    virtual void initUI() = 0;
    virtual void updateUI() = 0;
```